### PR TITLE
Allow users to signout if surveys are not available

### LINKED ISSUE #1879: Users with no available surveys get stuck on survey list
"Back" exits the app, so there's no way to log out. We should likely add a "Sign out" button and a message to the survey list screen if no surveys are available.

@shobhitagarwal1612 @JSunde FYI

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #1879

@gino-m  PTAL?

### CODE REVIEW

**Review by gino-m — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt [hunk 1]
```diff
@@ -45,11 +49,13 @@ internal constructor(
...
   private val activateSurveyUseCase: ActivateSurveyUseCase,
   @ApplicationScope private val externalScope: CoroutineScope,
-  @IoDispatcher private val ioDispatcher: CoroutineDispatcher
+  @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+  private val userRepository: UserRepository
 ) : AbstractViewModel() {
 
   val displayProgressDialog: MutableStateFlow<Boolean> = MutableStateFlow(false)
   val surveySummaries: Flow<List<SurveyItem>>
+  var surveyAvailable: LiveData<Boolean?>
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt]: Done

[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt]: Thanks! Done

File: ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt [hunk 2]
```diff
@@ -61,6 +67,7 @@ internal constructor(
             .sortedByDescending { it.isAvailableOffline }
         }
       }
+    surveyAvailable = surveySummaries.map { it.isNotEmpty() }.onStart { emit(true) }.asLiveData()
```
[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt]: We don't have a progress spinner when the surveys are refreshing. The initial true value prevents displaying the empty screen when the screen is loading.

[the remark this answers is not in the document]
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/surveyselector/SurveySelectorViewModel.kt]: This is now done. Thanks for the suggestions!

**Review by gino-m — APPROVED** [review]
LGTM!

**Pull request merged**